src/AutoVizuA11y.tsx

```diff
@@ -239,92 +219,113 @@ const AutoVizuA11y = ({
-						function (result) {
-							descsAux = result; // Output: [longerDescValue, smallerDescValue]
-							setDescs(result); // Output: [longerDescValue, smallerDescValue]
-							descriptionsChanger(ref, type, descsAux, title, autoDescriptions);
-
-							if (autoDescriptions && autoDescriptions.dynamicDescriptions === false) {
-								localStorage.setItem(storedLongerKey, descsAux[0]);
-								localStorage.setItem(storedSmallerKey, descsAux[1]);
-							}
-						},
-					);
+					generateDescriptions({
+						title,
+						data,
+						average: averageAux,
+						context,
+						apiKey: autoDescriptions!.apiKey,
+						model: autoDescriptions!.model,
+						temperature: autoDescriptions!.temperature,
+					}).then(function (result) {
+						chartDescriptions = result; // Output: [longerDescValue, smallerDescValue]
+						setDescs(result); // Output: [longerDescValue, smallerDescValue]
+						descriptionsKeyHandler({
+							chartRef,
+							setDescriptionContent,
+							type,
+							descs: chartDescriptions,
+							title,
+							autoDescriptions,
+						});
+
+						if (autoDescriptions && autoDescriptions.dynamicDescriptions === false) {
+							localStorage.setItem(storedLongerKey, chartDescriptions[0]);
+							localStorage.setItem(storedSmallerKey, chartDescriptions[1]);
+						}
+					});
 				}
 			});
 
-			//wipes the old tabindex present in the child components
-			levelChart(ref, true);
+			// sets the navigation onto the charts first
+			switchToChartLevel(chartRef, true);
 		}, 500);
+	}, [chartRef]);
 
-		//creates the ShortcutGuide
-		createShortcutGuide();
-	}, [ref]);
-
-	//sets the appropriate navigation keys in the ShortcutGuide
-	function handleNav(event: React.KeyboardEvent<HTMLDivElement>) {
-		levelNav(event, ref);
-	}
-
-	//sets the appropriate navigation keys and shortcuts in the charts and data
-	function handleKeyDown(event: React.KeyboardEvent<HTMLDivElement>) {
-		let alertDiv: HTMLElement | null = ref.current?.getElementsByClassName(
-			"a11y_alert",
-		)[0] as HTMLElement;
-		levelNav(event, ref, alertDiv, selectorType, multiSeries, nextSeries, series, selectedSeries);
-		let numberAux = keyHandler(
-			type,
-			event,
-			number,
-			ref,
-			selectorType,
-			insights,
-			insightsArray,
-			arrayConverted,
-			title,
-			descs,
-			series,
-			selectedSeries,
-			autoDescriptions,
-		);
-		setNumber(numberAux);
-	}
-
-	// features exclusive to bar charts (might be able to turn this more modular)
-	if (!selectorType) {
-		console.log("Type of chart not supported or no type given");
-	}
 	return (
-		<div
-			ref={ref}
-			onKeyDown={handleKeyDown}
-			className="a11y_chart"
-			data-testid="a11y_chart"
-			role="form"
-		>
-			<p
-				style={{ textIndent: "-10000px" }}
-				className="a11y_desc visually-hidden"
-				data-testid="a11y_desc"
-				onFocus={handleFocus}
-				onBlur={handleBlur}
+		<>
+			<div
+				ref={chartRef}
+				onKeyDown={(event) => {
```

Comment:
Suggestion: why not change this to a callback function in the component using useCallback that has a shape something like this:
```suggestion
const handleOnKeyDown = useCallback((event: React.KeyboardEvent<HTMLDivElement>) => {
  const DATA = {
		event,
		alertDivRef,
		type,
		number,
		chartRef,
		elements,
		selectedSeries,
		series,
		selectorType,
		setSelectedSeries,
		setNumber,
		setDescriptionContent,
		insights,
		insightsArray,
		arrayConverted,
		title,
		descs,
		autoDescriptions,
		multiSeries,
   };

   // This would have two parameters instead of one
   handleKeyDown(event, DATA);
}, [put-here-the-necessary-dependencies]);
...
onKeyDown={handleOnKeyDown}
```

This avoids having to recreate this expensive function everytime we re-render the component.